internal/controller/nimcache_controller.go

```diff
@@ -815,6 +798,19 @@ func (r *NIMCacheReconciler) reconcileNIMCache(ctx context.Context, nimCache *ap
 		logger.Error(err, "reconciliation of caching job failed", "job", getJobName(nimCache))
 		return ctrl.Result{}, err
 	}
+
+	obj := &appsv1alpha1.NIMCache{}
```

Comment:
Can we pull this into a separate function which does get and update of CR status.